src/RecordData/StorageType.h

```diff
@@ -0,0 +1,56 @@
+#ifndef STORAGE_TYPE_H
+#define STORAGE_TYPE_H
+
+namespace mmfs {
+
+class StorageType
+{
+public:
+    virtual ~StorageType() {};
+
+    /**
+     * Initialize the storage device.
+     * @return true if initialization was successful, false otherwise.
+     */
+    virtual bool init() = 0;
+
+    /**
+     * Check if the storage device is ready for read/write operations.
+     * @return true if the storage device is ready, false otherwise.
+     */
+    virtual bool isReady() const = 0;
+
+    /**
+     * Write data to the storage device.
+     * @param data The data to be written.
+     * @param size The size of the data to be written.
+     */
+    virtual void write(char *data, int size) = 0;
+
+    /**
+     * Read data from the storage device.
+     * @param data The buffer to store the read data.
+     * @param size The size of the data to be read.
+     * @return The number of bytes read. If unsuccessful, returns -1.
```

Comment:
I feel that this should return `0` if it fails, as it would be accurately 0 bytes read. I feel that nobody would ever try to read 0 bytes, so it makes slightly more semantic sense to return 0 if no bytes were read.

Reply:
this is more in line with how typical read functions and streams return stuff which I think helps avoid confusion, as -1 indicates an error opening while 0 indicates the thing was empty